### DISCUSSION TITLE
Testing Distributed Data Parallele with Pytest

### DISCUSSION THREAD

**Discussion by tchaton** [pytest-dev/pytest#7995] [Q&A]
Dear people from Pytest,

We are currently investigating testing our Distributed Data Parallel implementation in Pytorch Lightning (possibility with Pytest).

DPP launches one process per gpu and run deep learning training on multi-node multi-gpus.
Here is the line where we launch process using `subprocess.Popen(..)`
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/accelerators/ddp_accelerator.py#L127.

We wondered if Pytest Team would have knowledge about the best way to approach this problem. 

Best regards,
Thomas Chaton
Research Engineer at Pytorch Lightning.

**Comment by RonnyPfannschmidt** [comment]
its not even remotely clear whats actually being asked, its next to impossible to give a reasonable answer to such a seemingly complex topic without some more context for comprehension
